src/generators/default_map_generator_job.cpp

```diff
@@ -628,152 +707,53 @@ static void flood_name(const map_location& start, const std::string& name, std::
+	size_t width, size_t height, size_t island_size, size_t island_off_center, size_t iterations, size_t hill_size, size_t max_lakes, size_t nvillages, size_t castle_size, size_t nplayers,
+	bool roads_between_castles, std::map<map_location,std::string>* labels, const config& cfg)
 {
 	log_scope("map generation");
 
 	// Odd widths are nasty
 	VALIDATE(is_even(width), _("Random maps with an odd width aren't supported."));
 
-	/** Try to find configuration for castles. */
-	const config &castle_config = cfg.child("castle");
+	// Try to find configuration for castles
+	const config& castle_config = cfg.child("castle");
 	if(!castle_config) {
 		LOG_NG << "Could not find castle configuration\n";
 		return std::string();
 	}
 
 	int ticks = SDL_GetTicks();
 
-	// Find out what the 'flatland' on this map is, i.e. grassland.
-	std::string flatland = cfg["default_flatland"];
-	if(flatland == "") {
-		flatland = t_translation::write_terrain_code(t_translation::GRASS_LAND);
-	}
-
-	const t_translation::t_terrain grassland = t_translation::read_terrain_code(flatland);
-
-	// We want to generate a map that is 9 times bigger
-	// than the actual size desired.
-	// Only the middle part of the map will be used,
-	// but the rest is so that the map we end up using
-	// can have a context (e.g. rivers flowing from
-	// out of the map into the map, same for roads, etc.)
-	width *= 3;
+	// We want to generate a map that is 9 times bigger than the actual size desired.
+	// Only the middle part of the map will be used, but the rest is so that the map we
+	// end up using can have a context (e.g. rivers flowing from out of the map into the map,
+	// same for roads, etc.)
+	width  *= 3;
 	height *= 3;
 
-	LOG_NG << "generating height map...\n";
-	// Generate the height of everything.
-	const height_map heights = generate_height_map(width,height,iterations,hill_size,island_size,island_off_center);
-	LOG_NG << "done generating height map...\n";
-	LOG_NG << (SDL_GetTicks() - ticks) << "\n"; ticks = SDL_GetTicks();
+	config naming = game_config_.child("naming");
 
-	config naming;
-	naming.append_children(game_config_,"naming");
 	if(cfg.has_child("naming")) {
-		naming.append_children(cfg,"naming");
+		naming.append_children(cfg, "naming");
 	}
+
 	naming.merge_children("naming");
-	naming = naming.child_or_empty("naming");
 
 	// If the [naming] child is empty, we cannot provide good names.
 	std::map<map_location,std::string>* misc_labels = naming.empty() ? nullptr : labels;
 
-	std::shared_ptr<name_generator> base_name_generator;
-	std::shared_ptr<name_generator> river_name_generator;
-	std::shared_ptr<name_generator> lake_name_generator;
-	std::shared_ptr<name_generator> road_name_generator;
-	std::shared_ptr<name_generator> bridge_name_generator;
-	std::shared_ptr<name_generator> mountain_name_generator;
-	std::shared_ptr<name_generator> forest_name_generator;
-	std::shared_ptr<name_generator> swamp_name_generator;
+	std::shared_ptr<name_generator>
```

Comment:
It's nice to merge them into one statement, but what's the point if they're still each on their own line? At least put two or three on each line?